File: solutions/python/twelve-days/1/twelve_days.py
day_numbers = ["first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth"]

gifts = ["a Partridge in a Pear Tree.", "two Turtle Doves, ", "three French Hens, ", "four Calling Birds, ", "five Gold Rings, ", "six Geese-a-Laying, ", "seven Swans-a-Swimming, ", "eight Maids-a-Milking, ", "nine Ladies Dancing, ", "ten Lords-a-Leaping, ", "eleven Pipers Piping, ", "twelve Drummers Drumming, "]

def stanza(number):
    result = "On the " + day_numbers[number-1] + " day of Christmas my true love gave to me: "
    if number==1:
        result += gifts[0]
    else:
        for k in range(number-1,0,-1):
            result += gifts[k]
        result += "and " + gifts[0]
    return result
    
def recite(start_verse, end_verse):
    output = []
    for no in range(start_verse, end_verse+1):
        output.append(stanza(no))
    return output

File: solutions/java/twelve-days/1/src/main/java/TwelveDays.java
class TwelveDays {

    final String[] day_numbers =  {"first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth"};
    final String[] gifts = {"a Partridge in a Pear Tree",
"two Turtle Doves", "three French Hens", 
"four Calling Birds", "five Gold Rings", "six Geese-a-Laying", 
"seven Swans-a-Swimming", "eight Maids-a-Milking", 
"nine Ladies Dancing", "ten Lords-a-Leaping",
"eleven Pipers Piping", "twelve Drummers Drumming"};


    String verse(int number) {
        String all_gifts = "";
        String first_line = "On the " + day_numbers[number-1] + " day of Christmas my true love gave to me: ";
        if(number==1) 
            all_gifts+= gifts[0];
        else {        
            for(int idx = 1; idx<=number-1; idx++) {
                all_gifts = gifts[idx] + ", " + all_gifts; 
            }
            all_gifts += "and " + gifts[0];
        }
        return first_line + all_gifts + ".\n";
    }

    String verses(int from, int to) {
        String output = "";
        Boolean several = (from!=to);
        for(int i=from; i<=to; i++)
            output+= verse(i) + (several && i!=to? "\n" : "");
        return output;
    }
    
    String sing() {
        return verses(1,12);
    }
}
